odbc/src/api/data.rs

```diff
@@ -1078,6 +1078,30 @@ unsafe fn set_output_binary_helper(
 
 pub mod i16_len {
     use super::*;
+    ///
+    /// set_output_wstring_as_bytes writes [`message`] to the *WChar [`output_ptr`]. [`buffer_len`] is the
+    /// length of the [`output_ptr`] buffer in characters; the message should be truncated
+    /// if it is longer than the buffer length. The number of *BYTES* written to [`output_ptr`]
+    /// should be stored in [`text_length_ptr`].
+    ///
+    /// # Safety
+    /// This writes to multiple raw C-pointers
+    ///
+    pub unsafe fn set_output_wstring_as_bytes(
+        message: &str,
+        output_ptr: Pointer,
+        buffer_len: usize,
+        text_length_ptr: *mut SmallInt,
+    ) -> SqlReturn {
+        let message = message.encode_utf16().collect::<Vec<u16>>();
```

Comment:
If buffer_len is 1, we can return directly with an error because nothing will fit, no?

Reply:
we could, but `set_output_wstring_helper` is already handling that. I don't think the performance gain from this (roughly nil given how often we expect someone to pass a buffer size of 1 to a wchar function) justifies the increase in code complexity (slightly more than nil ;))